zephyr/lib/alloc.c

```diff
@@ -376,6 +376,20 @@ static void *heap_alloc_aligned(struct k_heap *h, size_t min_align, size_t bytes
 	struct sys_memory_stats stats;
 #endif
 
+	/*
+	 * Zephyr sys_heap stores metadata at start of each
+	 * heap allocation. To ensure no allocated cached buffer
+	 * overlaps the same cacheline with the metadata chunk,
+	 * align both allocation start and size of allocation
+	 * to cacheline. As cached and non-cached allocations are
+	 * mixed, same rules need to be followed for both type of
+	 * allocations.
+	 */
+#ifdef CONFIG_SOF_ZEPHYR_HEAP_CACHED
+	min_align = MAX(PLATFORM_DCACHE_ALIGN, min_align);
+	bytes = ALIGN_UP(bytes, min_align);
```

Comment:
I don't think we need to also align the size to the same value. If the user requests a page-aligned buffer, only the beginning should be page-aligned. The size can stay cache-line size aligned. Yes, usually if you request a page-aligned buffer the size is also a multiple of the page size, but for correctness / cache management we only need cache-line size alignment

Reply:
Agree. This was a copy-paste from the cached allocation implementation. Anyway, pushed a new version with size aligned to cache line.